lib/galaxy/security/validate_user_input.py

```diff
@@ -75,11 +84,23 @@ def validate_email(trans, email, user=None, check_dup=True, allow_empty=False):
         domain = extract_domain(email, base_only=True)
         if domain in trans.app.config.email_domain_blocklist_content:
             message = "Please enter your permanent email address."
+    if not message and validate_domain:
```

Comment:
```suggestion
    elif validate_domain:
```
Unless you think this check is conceptually different from the block above?

Reply:
Yeah, it is, because if this is an `elif` then this will be skipped if the allow or block lists are in use.

Reply:
You're right! (it's not conceptually - it's *logically* separate!)
That said, if the domain is invalid, the message will override the previous message (which is not incorrect - invalid is invalid, and an invalid domain doesn't make it "more" invalid, unlike "very invalid characters", as per PR description :)
However, I think that whole block of logic could be simplified by separating the steps:
```
if email string is invalid: return message
if domain is invalid: return message
if the email is OK but is blocked for other reasons (exists or is blocked or is supposed to be on the allowed list but is not): return message
```
What do you think? (I may be overthinking this)

Reply:
> That said, if the domain is invalid, the message will override the previous message

I think it doesn't because of `if not message and...`? However, your flow makes sense, I'll push an update.

Reply:
Oh, of course, you're right! The message will be set in a previous clause!